sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py

```diff
@@ -230,6 +232,7 @@ def _create_pipeline(self, credential, **kwargs):
             ContentDecodePolicy(),
             RedirectPolicy(**kwargs),
             StorageHosts(hosts=self._hosts, **kwargs),
+            StorageVersionCheckPolicy(),
```

Comment:
You can easily makes this compatible with all 3 servies by simply passing in kwargs here:
`StorageVersionCheckPolicy(**kwargs)`

Then add an init to your policy class:
```
class StorageVersionCheckPolicy:
    def __init__(self, storage_sdk, **kwargs):
        self.service = storage_sdk  # The value of this will be either 'blob', 'queue' or 'file-share'
```
You can see similar use of the service type in this policy:
https://github.com/Azure/azure-sdk-for-python/blob/1d396bc086d0e9ad91067909b3d9f57f30eb12a6/sdk/storage/azure-storage-blob/azure/storage/blob/_shared/policies.py#L266

Reply:
Updated in the second commit!